tests/test_data.py

```diff
@@ -347,6 +347,37 @@ async def test_environment_set_setting_parameter(init_dataclasses_and_load_schem
         await env.set(data.AUTO_DEPLOY, 5)
 
 
+async def test_environment_add_new_setting_parameter(init_dataclasses_and_load_schema):
+    project = data.Project(name="proj")
+    await project.insert()
+
+    env = data.Environment(name="dev", project=project.id, repo_url="", repo_branch="")
+    await env.insert()
+
+    new_setting: Setting = Setting(
+        name="a new boolean setting",
+        default=False,
+        typ="bool",
+        validator=convert_boolean,
+        doc="a new setting",
+    )
+
+    await env.register_setting(new_setting)
+    assert (await env.get("a new boolean setting")) is False
+    await env.set("a new boolean setting", True)
+    assert (await env.get(data.AUTO_DEPLOY)) is True
```

Comment:
I would verify this actions via the API instead.

Reply:
you mean using `await client.list_settings(tid=environment_default)`? why do you prefer that?

Reply:
Yes. More specifically `client.get_setting` and `client.set_setting`. It's better to use the API, because API's are intended to remain stable over time, which prevents changes to the test case in the future. Next to that it also turn it into an end-to-end test. And it's not really more work to use the API instead.